src/gc/gc.cpp

```diff
@@ -37212,6 +37212,24 @@ void GCHeap::DiagWalkObject (Object* obj, walk_fn fn, void* context)
     }
 }
 
+void GCHeap::DiagWalkObject2 (Object* obj, walk_fn2 fn, void* context)
+{
+    uint8_t* o = (uint8_t*)obj;
+    if (o)
+    {
+        go_through_object_cl (method_table (o), o, size(o), oo,
+                                    {
+                                        if (*oo)
+                                        {
+                                            Object *oh = (Object*)*oo;
+                                            if (!fn (obj, oh, context))
+                                                return;
+                                        }
+                                    }
+            );
+    }
```

Comment:
am I missing something? this looks exactly like the implementation of DaigWalkObject.